src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php

```diff
@@ -462,6 +462,12 @@ protected function forward(Request $request, $catch = false, Response $entry = n
         // is always called from the same process as the backend.
         $request->server->set('REMOTE_ADDR', '127.0.0.1');
 
+        // make sure HttpCache is a trusted proxy
+        if (!in_array('127.0.0.1', $trustedProxies = Request::getTrustedProxies())) {
```

Comment:
@stof this seems to be necessary also when using a subrequest (at least in my experience).